### PR TITLE
Add Tallycoin Connect

### PR DESCRIPTION
**Pull request by dennisreimann**
# App Submission

### App name

Tallycoin Connect

### Version

v1.7.0

### One line description of the app

Connect Tallycoin with your Umbrel node

### Summary of the app

Receive Tallycoin Lightning payouts directly to your own node.

### Developer name
djbooth007

### Developer website
https://github.com/djbooth007/tallycoin_connect

### Source code link
https://github.com/djbooth007/tallycoin_connect

### Support link
https://github.com/djbooth007/tallycoin_connect/issues

### Requires
- [ ] Bitcoin Core
- [ ] Electrum server
- [x] LND

### 256x256 SVG icon

https://svgur.com/s/b9D

### Gallery images
[see comment](https://github.com/getumbrel/umbrel/pull/1052#issuecomment-942779683)

### I have tested my app on:
- [x] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

Closes djbooth007/tallycoin_connect#13 djbooth007/tallycoin_connect#14

### CODE REVIEW

**Comment by lukechilds** [comment]
@dennisreimann nope, sorry been relocating so I'm a bit behind, will review this as soon as I'm able, looking forward to having Tallycoin Connect in Umbrel!

File: apps/tallycoin-connect/data/tallycoin_api.key [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (lukechilds) [inline apps/tallycoin-connect/data/tallycoin_api.key:1]: Is this file needed if it's empty?

File: apps/tallycoin-connect/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    restart: on-failure
+    stop_grace_period: 1m
+    ports:
+      - "$APP_TALLYCOIN_CONNECT_PORT:$APP_TALLYCOIN_CONNECT_PORT"
+    volumes:
+      - ${LND_DATA_DIR}:/lnd:ro
+      - ${APP_DATA_DIR}/data:/data
+    environment:
+      LND_SOCKET: "$LND_IP:$LND_GRPC_PORT"
+      LND_MACAROON_PATH: "/lnd/data/chain/bitcoin/$BITCOIN_NETWORK/admin.macaroon"
```
Comment (lukechilds) [inline apps/tallycoin-connect/docker-compose.yml:15]: Do we need to pass in the admin macaroon here or would the invoice macaroon be sufficient?

Comment (djbooth007) [reply apps/tallycoin-connect/docker-compose.yml:15]: I believe the dependency ln-service requires the admin macaroon, although Tallycoin Connect does not require (or use) full permissions.

**Comment by djbooth007** [comment]
Umbrel doesn't need to be publicly accessible. Tallycoin Connect will try connect to the server over websocket port 8123 (default), unless a different port was specified during setup. Is the port open in the firewall?

**Comment by lukechilds** [comment]
>Is the port open in the firewall?

Do you mean on the router's firewall or Umbrel's firewall? The port is open on my Umbrel, no port forwarding is setup for my router, that was what I wasreferring to about if the Umbrel needing to be publically accessible.

Maybe I'm misunderstanding but if the Umbrel is making an outbound connection from Tallycoin Connect to the Tallycoin server it should not need manual port forwarding rules setup to work. It should just connect out normally with no inbound port forwarding rules required. Am I misunderstanding how this works?

Have you guys tested this on an Umbrel or dev environment and been able to pair successfully?

File: apps/tallycoin-connect/docker-compose.yml (hunk 2 above)
Comment (lukechilds) [reply apps/tallycoin-connect/docker-compose.yml:15]: Ok got it!

This doesn't need to be a blocker for the initial release, it's fine to as is, but it would be great to see a future version of Tallycoin Connect that works with just an invoice macaroon. 😍

File: apps/tallycoin-connect/data/tallycoin_api.key (hunk 1 above)
Comment (dennisreimann) [reply apps/tallycoin-connect/data/tallycoin_api.key:1]: I think it couldn't get written if it wasn't present already -- iirc that's the reason why I added this empty version.

Comment (lukechilds) [reply apps/tallycoin-connect/data/tallycoin_api.key:1]: Ahh got it. Where you able to connect the app with the server ok? I seem to be having issues.

**Comment by djbooth007** [comment]
Probably not your misunderstanding and more about my poor explanation! 
Users shouldn't need to setup port forwarding on their router, however some node implementations block a range of in/outgoing ports for security. So just wanted to mention this in case you have firewalled traffic on the node itself.

@dennisreimann may have more insight into testing for Umbrel specifically. I double-checked the server and it's OK.

**Comment by lukechilds** [comment]
>Probably not your misunderstanding and more about my poor explanation!
Users shouldn't need to setup port forwarding on their router, however some node implementations block a range of in/outgoing ports for security. So just wanted to mention this in case you have firewalled traffic on the node itself.

We use Docker in Umbrel so every app is running in a container, not directly on the host. Docker will by default not expose any ports on the container via the host unless we explicitly enable it. The current configuration in this PR exposes port 8123 on the container as port 8123 on the host. Any other ports on the Tallycoin Connect container will only accessable inside the private internal Docker network. So that sounds like everything should be setup in order for the app to connect.

**Comment by dennisreimann** [comment]
@lukechilds I'm not sure whether or not I tested pairing with the API, but can do that as soon as I find the time.

**Comment by lukechilds** [comment]
Hmmn, weird, made some changes and reinstalled and now it's all working. I didn't change anything that should have affected anything with the connection so not quite sure what was going wrong before but it's all working now! 🎉

**Review by lukechilds — APPROVED** [review]

**Comment by dennisreimann** [comment]
@lukechilds Glad to hear it's working now! Looking forward to see this in Umbrel :)

**Comment by lukechilds** [comment]
Merged 🚀

Thanks for this @dennisreimann @djbooth007!

**Comment by lukechilds** [comment]
Oh also guys currently we have the description as:

>Receive Tallycoin Lightning payouts directly to your own node.

Umbrel users not be aware of what Tallycoin is when they view the app in the app store. It might be a good idea to add a paragraph explaining what Tallycoin is and why it's useful. And then explain this app lets you connect Tallycoin directly to your Umbrel LND node.

If you can give me a short description on Tallycoin I'll get the registry updated.

**Comment by dennisreimann** [comment]
Let's borrow what they use on their own website:

> Tallycoin is a crowdfunding platform built on top of Bitcoin blockchain. 
> Every satoshi donated to you, belongs to you. Tallycoin takes no cut.

As for the password: I can add the plaintext version and update Tallycoin Connect, but I'm busy with other things right now. WIll try to fit it in soon.

**Comment by lukechilds** [comment]
Done, thanks @dennisreimann! https://github.com/getumbrel/umbrel/commit/77f4dafacf80f9945d48a5ffef21114f585d5dbc

>As for the password: I can add the plaintext version and update Tallycoin Connect, but I'm busy with other things right now. WIll try to fit it in soon.

Of course! No rush, Tallycoin Connect already allows users to manually set a password so it's totally shippable as is.

**Comment by mayankchhabra** [comment]
Great work! Congrats on the launch in the app store, @djbooth007 and @dennisreimann! :)

There were some rendering issues with the app icon's SVG so we tried to recreate it:



And also tweaked the gallery images slightly:





If you'd like to change any of the images or the app icon (now or in the future), feel free to directly open a PR at [getumbrel/umbrel-apps-gallery](https://github.com/getumbrel/umbrel-apps-gallery).

**Comment by dennisreimann** [comment]
Nice, looking good – thank you too! 🙏

**Comment by djbooth007** [comment]
Awesome! Thanks so much 💯
